docs/source/usage.rst

```diff
@@ -0,0 +1,230 @@
+   be setup for the registered model.
+3. ``eav_attr`` - Named of the Entity toolkit instance on the registered
+   model instance. "eav" by default. See attach_eav_attr.
+4. ``generic_relation_attr`` - Name of the GenericRelation to Value
+   objects. "eav_values" by default.
+5. ``generic_relation_related_name`` - Name of the related name for
+   GenericRelation from Entity to Value. None by default. Therefore,
+   if not overridden, it is not possible to query Values by Entities.
+
+Example registration may look like:
+
+.. code-block:: python
+
+    class SupplierEavConfig(EavConfig):
+        manager_attr = 'eav_objects'
+
+    eav.register(supplier, SupplierEavConfig)
+
+Additionally, :class:`~eav.registry.EavConfig` provides *classmethod*
+:meth:`~eav.registry.EavConfig.get_attributes` which is used to determine
+a set of attributes available to a given model. By default, it returns
+``Attribute.objects.all()``. As usual, it can be customized:
+
+.. code-block:: python
+
+    class SomeModelEavConfig(EavConfig):
+        @classmethod
+        def get_attributes(cls):
+            return Attribute.objects.filter(slug__startswith='a')
+
+Attribute validation includes checks against illegal attribute value
+assignments. This means that value assignments for attributes which are
+excluded for the model are treated with
+:class:`~eav.exceptions.IllegalAssignmentException`. For example (extending
+previous one):
+
+.. code-block:: python
+
+    some_model.eav.beard = True
+    some_model.save()
+
+will throw an exception.
+
+Creating Attributes
+-------------------
+
+Once your models are registered, you can starting creating attributes for
+them. Two most important attribute attributes (huh?) are *slug* and
+*datatype*. *slug* is a unique identifier and must be a valid Python
+variable name, as it's used to access values for that attribute from
+:class:`~eav.models.Entity` helper:
+
+.. code-block:: python
+
+    Attribute.objects.create(slug='color', datatype=Attribute.TYPE_TEXT)
+    flower.eav.color = 'red'
+
+    # Alternatively, assuming you're using default EntityManager:
+    Flower.objects.create(name='rose', eav__color='red')
+
+*datatype* determines type of attribute (and by extension type of value
+stored in :class:`~eav.models.Value`). Available choices are:
+
+========= ==================
+  Type    Attribute Constant
+========= ==================
+*int*     ``TYPE_INT``
+*float*   ``TYPE_FLOAT``
+*text*    ``TYPE_TEXT``
+*date*    ``TYPE_DATE``
+*bool*    ``TYPE_BOOLEAN``
+*object*  ``TYPE_OBJECT``
+*enum*    ``TYPE_ENUM``
+========= ==================
+
+If you want to create an attribute with data-type *enum*, you need to provide
+it with ``enum_group``:
+
+.. code-block:: python
+
+    true = EnumValue.objects.create(value='Yes')
+    false = EnumValue.objects.create(value='No')
+    bool_group = EnumGroup.objects.create(name='Yes / No')
+    bool_group.enums.add(true, false)
```

Comment:
This might be a remark to the lib API and not really docs, and if so, let's make an issue out it for further discussion, but here it goes anyway.

I don't really understand (and hence like) the naming here.

`true` and `false` are `EnumValue`s, so a thing allowing to choose from either this or that, is logically an enum (with two choices possible). However, you then create something called `EnumGroup` - the name implies it consists of multiple Enums, so multiple things with multiple choices. But no, you then add **values** to a property called `enums`.

I'm lost on the relation between `EnumValue`, `EnumGroup` and **enum** as a logical being. The naming is confusing to me. How I understand this, it could go as follows:

```
true = EnumValue.objects.create(value='Yes')
false = EnumValue.objects.create(value='No')
bool_enum = Enum.objects.create(name='Yes / No')
bool_enum.values.add(true, false)
```

and it would be much more self-explanatory. I know this might be legacy, but something to keep on the radar.

Reply:
It is legacy. I assumed "enums" stands for "enum values".